README.md

```diff
@@ -53,10 +53,199 @@ Myget feeds:
+
+In the examples below we demonstrate how to create and enrich spans though OpenTelemetry API.
+
+OpenTelemetry also provides auto-collectors for ASP.NET Core, HttpClient calls (.NET Core) and Azure SDKs - [configuration](#configuration) section demonstrates how to enable it.
+
+### Obtaining tracer
+
+**Applications** should follow [configuration](#configuration) section to find out how to create/obtain tracer.
+
+**Libraries** must take dependency on OpenTelemetry API package only and should never instantiate tracer or configure OpenTelemetry. Libraries **will** be able to obtain *global* tracer that may either be noop (if user application is not instrumented with OpenTelemetry) or real tracer implemented in the SDK package.
+
+### Create basic span
+
+To create the most basic span, you only specify the name. OpenTelemetry SDK collects start/end timestamps, assigns tracing context and assumes status of this span is `OK`.
+
+```csharp
+var span = tracer
+    .SpanBuilder("basic span")
+    .StartSpan();
+
+span.End();
+```
+
+### Create nested spans
+
+In many cases you want to collect nested operations. You can propagate parent spans explicitly in your code or use implicit context propagation embedded into OpenTelemetry and .NET.
+
+#### Explicit parent propagation and assignment
+
+```csharp
+var parentSpan = tracer
+    .SpanBuilder("parent")
+    .StartSpan();
+
+var childSpan = tracer
+    .SpanBuilder("child")
+    .SetParent(parentSpan) // explicitly assigning parent here
+    .StartSpan();
+
+childSpan.End();
+parentSpan.End();
+```
+
+#### Implicit parent propagation and assignment
+
+```csharp
+var parentSpan = tracer
+    .SpanBuilder("parent")
+    .StartSpan();
+
+// calling WithSpan puts parentSpan into the ambient context
+// that flows in async calls.   When child is created, it
+// implicitly becomes child of current span
+using (tracer.WithSpan(parentSpan))
+{
+    var childSpan = tracer
+        .SpanBuilder("child")
+        .StartSpan();
+
+    childSpan.End();
+}
+```
+
+### Span with attributes
+
+Attributes provide additional context on span specific to specific operation it tracks such as HTTP/DB/etc call properties.
+
+```csharp
+var span = tracer
+    .SpanBuilder("span with attributes")
+    // spans have Client, Server, Internal, Producer and Consumer kinds to help visualize them
+    .SetSpanKind(SpanKind.Client)
+    .StartSpan();
+
+// attributes specific to the call
+span.SetAttribute("db.type", "redis");
+span.SetAttribute("db.instance", "localhost:6379[0]");
+span.SetAttribute("db.statement", "SET");
+span.End();
+```
+
+### Span with links
+
+Links allows to create relationships between different traces i.e. allow spans to have multiple relatives. They are typically used to tracer batching scenarios where multiple traces are merged into another one.

```

Comment:
link to spec? (https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/overview.md#links-between-spans)